server/error_recorder.go

```diff
@@ -0,0 +1,55 @@
+package server
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/bytebase/bytebase/api"
+	"github.com/labstack/echo/v4"
+)
+
+func errorRecorderMiddleware(s *Server, next echo.HandlerFunc) echo.HandlerFunc {
+	return func(c echo.Context) (err error) {
+		defer func() {
+			if c.Response().Status == http.StatusInternalServerError {
+				req := c.Request()
+				// get error message and stack trace if any
+				errorMessage := c.Get("errorMessage")
```

Comment:
can we get these info from c.Response()?

or at least we can get it via c.Error() instead of injecting our own key